### PR TITLE
WT-7736 Return an error when attempting to import a tiered table.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

**Comment by sueloverso** [comment]
Making a PR but not yet ready for review as the new test does not pass. Probably a test issue, but working on it.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

**Review by ddanderson — APPROVED** [review]
Looks good - a couple minor points.

File: src/block/block_read.c [hunk 1]
```diff
@@ -27,6 +27,13 @@ __wt_bm_read(
       session, block, addr, addr_size, &objectid, &offset, &size, &checksum));
 
 #ifdef HAVE_DIAGNOSTIC
+    /*
+     * In diagnostic mode, verify the user isn't trying to import a tiered object as a regular table
+     * file.
+     */
+    if (objectid != 0 && !WT_PREFIX_MATCH(block->name, "tiered:") &&
+      !WT_SUFFIX_MATCH(block->name, ".wtobj"))
+        WT_RET_MSG(session, ENOTSUP, "%s: is a tiered object", block->name);
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply src/block/block_read.c:36]: Given it is on the read path, and importing is very rare, and importing objects is very rare and trying to do this one bad use case, it seemed reasonable to put it under diagnostic (which was already there doing other checking too).

**Pull request merged**